heartbeat/beater/manager.go

```diff
@@ -231,7 +233,13 @@ func (m *monitor) Update(configs []*common.Config) error {
 		}
 
 		job := t.createJob(client)
-		jobCancel := m.manager.jobControl.Add(t.config.Schedule, id, job)
+		jobCancel, err := m.manager.jobControl.Add(t.config.Schedule, id, job)
+		if err == scheduler.ErrAlreadyStopped {
+			logp.Info("Will not add new task to scheduler. Heartbeat is stopping.")
+		} else if err != nil {
+			panic(pkgerrors.Wrap(err, "an unexpected error occurred adding a task to the scheduler: "))
```

Comment:
do we really want to have a panic here, meaning stopping heartbeat. Based on the current code path I think it's not possible but what if someone modifies jobControl and returns an additional error type?

Reply:
Good point. Maybe I'll just make this a `logp.Error`?